src/lua/datetime.lua

```diff
@@ -0,0 +1,611 @@
+    local dt = DAYS_EPOCH_OFFSET
+
+    local y = obj.year
+    if y ~= nil then
+        check_range(y, 1, 9999, 'year')
+        ymd = true
+    end
+    local M = obj.month
+    if M ~= nil then
+        check_range(M, 1, 12, 'month')
+        ymd = true
+    end
+    local d = obj.day
+    if d ~= nil then
+        check_range(d, 1, 31, 'day', -1)
+        ymd = true
+    end
+    local h = obj.hour
+    if h ~= nil then
+        check_range(h, 0, 23, 'hour')
+        hms = true
+    end
+    local m = obj.min
+    if m ~= nil then
+        check_range(m, 0, 59, 'min')
+        hms = true
+    end
+    local s = obj.sec
+    if s ~= nil then
+        check_range(s, 0, 60, 'sec')
+        hms = true
+    end
+
+    local nsec, usec, msec = obj.nsec, obj.usec, obj.msec
+    local count_usec = bool2int(nsec ~= nil) + bool2int(usec ~= nil) +
+                       bool2int(msec ~= nil)
+    if count_usec > 1 then
+        error('only one of nsec, usec or msecs may defined simultaneously', 2)
+    end
+    if usec ~= nil then
+        nsec = usec * 1e3
+    elseif msec ~= nil then
+        nsec = msec * 1e6
+    end
+    local ts = obj.timestamp
+    if ts ~= nil then
+        local fraction
+        s, fraction = math_modf(ts)
+        -- if there are separate nsec, usec, or msec provided then
+        -- timestamp should be integer
+        if count_usec == 0 then
+            nsec = fraction * 1e9
+        elseif fraction ~= 0 then
+            error('only integer values allowed in timestamp '..
+                  'if nsec, usec, or msecs provided', 2)
+        end
+        hms = true
+    end
+
+    local offset = obj.tzoffset
+    if offset ~= nil then
+        offset = get_timezone(offset)
+        check_range(offset, -720, 720, offset)
+    end
+
+    if obj.tz ~= nil then
+        nyi('tz')
+    end
+
+    -- .year, .month, .day
+    if ymd then
+        y = y or 1970
+        M = M or 1
+        d = d or 1
+        if d < 0 then
+            d = builtin.tnt_dt_days_in_month(y, M)
+        end
+        dt = builtin.tnt_dt_from_ymd(y, M, d)
+    end
+
+    -- .hour, .minute, .second
+    local secs = 0
+    if hms then
+        secs = (h or 0) * 3600 + (m or 0) * 60 + (s or 0)
+    end
+
+    return datetime_new_dt(dt, secs, nsec, offset or 0)
+end
+
+--[[
+    Convert to text datetime values
+
+    - datetime will use ISO-8601 format:
+        1970-01-01T00:00Z
+        2021-08-18T16:57:08.981725+03:00
+]]
+local function datetime_tostring(self)
+    check_date(self, 'datetime.tostring()')
+
+    local buff = ffi.new('char[?]', TOSTRING_BUFSIZE)
+    local len = builtin.datetime_to_string(buff, TOSTRING_BUFSIZE, self)
+    assert(len < TOSTRING_BUFSIZE)
+    return ffi.string(buff)
+end
+
+--[[
+    Dispatch function to create datetime from string or table.
+    Creates default timeobject (pointing to Epoch date) if
+    called without arguments.
+]]
+local function datetime_from(o)
+    if o == nil or type(o) == 'table' then
```

Comment:
Yes, this is extension to the RFC, as it was implemented in prior patch. This will be further extended at the next commit when parsing will be added, so 
```
date = require('datetime')
T = date('2021-09-01')
```
will work conveniently. This is simply preparation, providing reasonable default behaviour.

Reply:
But it is not reasonable now - you ignore errors. If I pass `date(100)`, it just does not return anything. Please, try not to extend RFC until we finish the basic version. It does not speed up the work on the patch really.